src/main/scala/nl.knaw.dans.easy.bagstore/server/ServletEnhancedLogging.scala

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.bagstore.server
+
+import javax.servlet.http.HttpServletRequest
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.scalatra.{ ActionResult, ScalatraBase }
+
+import scala.util.{ Failure, Success, Try }
+
+// TODO candidate for dans-scala-lib (copied from easy-deposit-api)
+trait ServletEnhancedLogging extends DebugEnhancedLogging {
+  this: ScalatraBase =>
+
+  before() {
+    logger.info(s"${ request.getMethod } ${ request.getRequestURL } remote=${ request.getRemoteAddr } params=$params headers=${ request.headers }")
```

Comment:
If a manual test shows unwanted info, apply a filter like:
```suggestion
    val headers = request.headers
      .withFilter { case (key, _) => key != "Authorization" }
      .map { case (key, value) => s"$key -> $value" }.mkString(", ")
    logger.info(s"${ request.getMethod } ${ request.getRequestURL } remote=${ request.getRemoteAddr } params=$params headers=Map(${ headers })")
```

Reply:
If we're happy with this logger, we should probably add it to dans-scala-lib. It's getting copied too many times already...

But maybe you can make some hooks for unwanted information etcetera?

Reply:
I'm afraid it is not yet mature enough. For example I don't see responses logged in this project. I remember from other projects that it required some discipline in writing the servlets and it might be a wrong approach at all.

Reply:
I'm not sure we should log the response body itself. That's sometimes quite a lot of data and that is typically a part that is likely to contain personal data...

Reply:
Manually logging the response header with the current API seems reasonable to me. But maybe you're right in that we should look somewhat more into the problem space.

Reply:
How many projects do you think it is copied? Is far as I can track I started with [deposit-api], copied it here and had some work in progress (not even committed) lying around for [solr4files-index].

[solr4files-index]: https://github.com/DANS-KNAW/easy-solr4files-index  
[deposit-api]: https://github.com/DANS-KNAW/easy-deposit-api 

Logging request seems stable. But if you want hooks for filtering I would not recommend a library component for a trait with nothing more than four lines at the start of this discussion.

Reply:
Our messages crossed. I didn't try to log the response body, just the response-code, headers and repeating the request method as a kind of bracketing.

Reply:
> How many projects do you think it is copied?
I copied it in the last 2 weeks to narcis-pid-aggregator, narcis-pid-manager and I think @PaulBoon copied it from there to narcis-pid-graph.

That's a total of 6 now...
When you get back from holiday, we should definitely do something about that! @jo-pol Can you make a JIRA issue for it?